protocol/x/clob/abci.go

```diff
@@ -197,8 +197,26 @@ func PrepareCheckState(
 	}
 
 	// 6. Get all potentially liquidatable subaccount IDs and attempt to liquidate them.
-	subaccountIds := daemonLiquidationInfo.GetLiquidatableSubaccountIds()
-	if err := keeper.LiquidateSubaccountsAgainstOrderbook(ctx, subaccountIds); err != nil {
+	liquidatableSubaccountIds := daemonLiquidationInfo.GetLiquidatableSubaccountIds()
+	subaccountsToDeleverage, err := keeper.LiquidateSubaccountsAgainstOrderbook(ctx, liquidatableSubaccountIds)
+	if err != nil {
+		panic(err)
+	}
+	subaccountPositionInfo := daemonLiquidationInfo.GetSubaccountsWithPositions()
```

Comment:
heads up - I am planning to update `GetSubaccountsWithPosition` to take a `perpetualId` and also `isLong` as parameters. 

currently GetSubaccountsWithPositions makes a copy of the entire map, which is not necessary in most cases. 

non blocking, feel free to merge and I can update this in my PR